holonote/annotate/annotator.py

```diff
@@ -317,15 +408,28 @@ def commit(self, return_commits=False):
             return commits
 
 
-class AnnotatorPlot(AnnotatorInterface):
+class AnnotatorElement(param.Parameterized):
```

Comment:
I don't understand this concept of an `AnnotatorElement` and how it is distinct from an `Annotator`. Whatever it is, it is not a holoviews element so I think this is confusing.

Reply:
I've just noticed the `kdims`... once `rect_min` and `rect_max` are gone, isn't that the key feature of this class?

Reply:
No, the key feature of the class is that the `Annotator` can have multiple `AnnotatorElement` associated with it. A small sketch of how I see the internal (note that some changes are still needed in follow-up PRs). The dashed line should be seen as something a normal user should not interact with.

![1](https://github.com/holoviz/holonote/assets/19758978/0930705f-97ee-4cb4-9120-ba0f08097bb2)

As for the naming itself, I think it is fine, as I see it as a provider of an element to the annotator. In the same way, `AnnotationTable` is a provider of the tables even though it is not directly a table.